ledger-service/http-json/src/main/scala/com/digitalasset/http/Main.scala

```diff
@@ -110,30 +109,30 @@ object Main {
     val contractDao = config.jdbcConfig.map(c => ContractDao(c.driver, c.url, c.user, c.password))
 
     (contractDao, config.jdbcConfig) match {
-      case (Some(dao), Some(c)) if c.createSchema =>
-        logger.info("Creating DB schema...")
-        import dao.{logHandler, jdbcDriver}
-        Try(dao.transact(ContractDao.initialize).unsafeRunSync()) match {
-          case Success(()) =>
-            logger.info("DB schema created. Terminating process...")
-            terminate()
-            System.exit(ErrorCodes.Ok)
+      case (Some(dao), Some(c)) =>
+        def terminateProcess(): Unit = {
+          logger.info("Terminating process...")
+          terminate()
+          System.exit(ErrorCodes.Ok)
+        }
+        Try(SchemaHandlingResult.fromSchemaHandling(dao, c.schemaHandling).unsafeRunSync()) match {
+          case Success(SchemaHandlingResult.Terminate) =>
+            terminateProcess()
+          case Success(SchemaHandlingResult.Continue) => ()
           case Failure(e) =>
-            logger.error("Failed creating DB schema", e)
-            terminate()
-            System.exit(ErrorCodes.StartupError)
+            logger.error("Failed processing the schema handling", e)
```

Comment:
this sounds a bit weird, I think we should say something like "failed to create the schema", "schema version mismatch, expected X got Y", …..

Reply:
I improved the logging on the processing side such that any additional logging at this location is not necessary anymore, see 34669f7